### ISSUE TITLE
API is down

### ISSUE THREAD

**Issue by DeadlyEmbrace** [croberts22/umalapi#8]
Get the following error when trying to retrieve any anime:
![apierror](https://cloud.githubusercontent.com/assets/2103860/7458271/68727552-f293-11e4-98e3-830457721a37.JPG)
Have tried rolling back to an earlier version but the problem remains. Suspect that something on one of the pages has changed. Error can be reproduced by going to any anime link.

Unfortunately my Ruby skills isn't very good and I don't have time to play with it right now.

**Comment by croberts22** [comment]
Thanks for letting me know. HostGator is probably either down, or Crave's whitelisting service is unavailable. Pretty frustrating since they require specific IP addresses versus a domain.

**Comment by DeadlyEmbrace** [comment]
Yeah, that is a rather frustrating way of handling it.
Seems to be back up this morning so probably Crave's whitelisting service was down.

**Comment by croberts22** [comment]
Funny you mention that just now, I’m talking to HostGator right now in resolving some of the issues I’ve been having. The Ruby app seems to spawn multiple instances when it doesn’t get used in a while, and they don’t terminate themselves since they’re always waiting for a new request to come in. Unfortunately, HostGator doesn’t have any nice way for me to terminate those myself unless I request live chat support to terminate them. 




I’ve been in the process for looking for a new service to run the API on, probably Heroku. The only issue is dealing with Crave’s whitelisting service at that point, since the API would have a dynamic IP coming in. A static IP would cost a pretty penny per month.




Companies make things so difficult sometimes. :/

**Comment by DeadlyEmbrace** [comment]
My instance is actually hosted on Heroku. I've dug through my emails just now to look for my original key request and it seems when I applied for my API Key they allowed you to provide a domain and specify that the IP is dynamic. I wonder if they have changed the policy in the meantime? (I haven't checked their API docs in ages really)

Another option might be to host it in the Azure cloud (though you'll need to setup and manage the entire VM and it is fairly costly) as they hold on to the same external IP as long as they aren't shut down.

Yeah, it's really sad that we have to jump through all of these hoops just to make something simple work. Would have been a lot easier if their API access worked in a way similiar to AniDB. :/

**Comment by croberts22** [comment]
Interesting. Yeah, I did the same as well, but after sending a number of requests, I got the same error you encountered. So I contacted them, and they asked for a specific IP address. It was only until then that it worked "reliably," and then my instances on HostGator would get terminated. Sigh. 




But yeah, their docs haven't changed. I tried contacting them (Crave) again with no response. Not sure if they're tired of my emails or what. XD haha.


But my next project is to move it to Heroku, and then rewrite it in Python/Django. I'm not super familiar with Ruby in the first place, and I've been meaning to write a RESTful API in Python anyway.

**Comment by DeadlyEmbrace** [comment]
Ugh... really doesn't help that they are so unhelpful. To an extent I can understand where they are coming from but it really makes it hard on the devs who want to do something amazing.

Ooohhh.... I've recently started with Python (so not an expert) but I'm willing to lend a hand where I can. :)

**Comment by croberts22** [comment]
Yeah. Honestly I wish they would open up their API for development. That thing hasn't been updated in years.

I'll need all the help I can get! I messed with Python and Django back in college for a semester, and it was pretty nice. I hope to follow a similar pattern and have an extensive test suite for it. The Ruby project doesn't really have any tests, which always makes it worrisome whenever something breaks. :/ I'd like to follow something similar to the [Atarashii API](https://bitbucket.org/ratan12/atarashii-api), except in Python. I had considered helping out on that project, but I've already done my time in PHP. It's not my favorite language. ;)

**Comment by DeadlyEmbrace** [comment]
Kinda sad when big companies refuse to spend money on something that would be fairly trivial to implement. Oh well, guess we just have to make the most of what we got.

I agree, an extensive test suite is really necessary. I've only recently started messing with Python a little bit, but I guess I'll pick it up quickly. The Atarashii API looks quite nice but I agree, I'd prefer not to serve another durance under PHP, I'd rather use nicer languages for the code I write in my spare time. :D
